test/functional/wallet_bumpfee.py

```diff
@@ -165,9 +169,81 @@ def test_invalid_parameters(self, rbf_node, peer_node, dest_address):
         assert_raises_rpc_error(-8, "Invalid parameter, duplicate key: data",
                 rbf_node.bumpfee, rbfid, {"outputs": [{"data": "deadbeef"}, {"data": "deadbeef"}]})
 
+        self.log.info("Test reduce_output option")
+        assert_raises_rpc_error(-1, "JSON integer out of range", rbf_node.bumpfee, rbfid, {"reduce_output": -1})
+        assert_raises_rpc_error(-8, "Change position is out of range", rbf_node.bumpfee, rbfid, {"reduce_output": 2})
+
+        self.log.info("Test outputs and reduce_output cannot both be provided")
+        assert_raises_rpc_error(-8, "outputs and reduce_output cannot both be specified simultaneously", rbf_node.bumpfee, rbfid, {"reduce_output": 2, "outputs": [{dest_address: 0.1}]})
+
         self.clear_mempool()
 
 
+    def test_provided_change_pos(self, rbf_node):
+        self.log.info("Test the reduce_output option")
+
+        change_addr = rbf_node.getnewaddress()
+        dest_addr = rbf_node.getnewaddress()
+        assert_equal(rbf_node.getaddressinfo(change_addr)["ischange"], False)
+        assert_equal(rbf_node.getaddressinfo(dest_addr)["ischange"], False)
+
+        send_res = rbf_node.send(outputs=[{dest_addr: 1}], options={"change_address": change_addr})
+        assert send_res["complete"]
+        txid = send_res["txid"]
+
+        tx = rbf_node.gettransaction(txid=txid, verbose=True)
+        assert_equal(len(tx["decoded"]["vout"]), 2)
+
+        change_pos = find_vout_for_address(rbf_node, txid, change_addr)
+        change_value = tx["decoded"]["vout"][change_pos]["value"]
+
+        bumped = rbf_node.bumpfee(txid, {"reduce_output": change_pos})
+        new_txid = bumped["txid"]
+
+        new_tx = rbf_node.gettransaction(txid=new_txid, verbose=True)
+        assert_equal(len(new_tx["decoded"]["vout"]), 2)
+        new_change_pos = find_vout_for_address(rbf_node, new_txid, change_addr)
```

Comment:
Will `new_change_pos == change_pos` always here? Or are the outputs of the replacement re-randomized?

Reply:
They will be re-randomized.